cmd/nerdctl/main.go

```diff
@@ -87,17 +88,41 @@ func xmain() error {
 // Config corresponds to nerdctl.toml .
 // See docs/config.md .
 type Config struct {
-	Debug            bool     `toml:"debug"`
-	DebugFull        bool     `toml:"debug_full"`
-	Address          string   `toml:"address"`
-	Namespace        string   `toml:"namespace"`
-	Snapshotter      string   `toml:"snapshotter"`
-	CNIPath          string   `toml:"cni_path"`
-	CNINetConfPath   string   `toml:"cni_netconfpath"`
-	DataRoot         string   `toml:"data_root"`
-	CgroupManager    string   `toml:"cgroup_manager"`
-	InsecureRegistry bool     `toml:"insecure_registry"`
-	HostsDir         []string `toml:"hosts_dir"`
+	Debug            bool         `toml:"debug"`
+	DebugFull        bool         `toml:"debug_full"`
+	Address          string       `toml:"address"`
+	Namespace        string       `toml:"namespace"`
+	Snapshotter      string       `toml:"snapshotter"`
+	CNIPath          string       `toml:"cni_path"`
+	CNINetConfPath   string       `toml:"cni_netconfpath"`
+	DataRoot         string       `toml:"data_root"`
+	CgroupManager    string       `toml:"cgroup_manager"`
+	InsecureRegistry bool         `toml:"insecure_registry"`
+	HostsDir         []string     `toml:"hosts_dir"`
+	RunConfig        *RunConfig   `toml:"default_run_config"`
+	BuildConfig      *BuildConfig `toml:"default_build_config"`
+}
+
+type RunConfig struct {
+	Platform string   `toml:"platform"`
+	Network  []string `toml:"network"`
+}
```

Comment:
Can we reuse the compose structure? https://pkg.go.dev/github.com/compose-spec/compose-go@v1.2.8/types#ServiceConfig

(Some fields such as `image` and `container_name` cannot be supported and should raise an error.)

Reply:
I think we should not reuse https://pkg.go.dev/github.com/compose-spec/compose-go@v1.2.8/types#ServiceConfig. 

I think this would be a little bit complicated for user. For instance, like #1128, maybe people just need config like this

```toml
[default_run_config]
network=["test"]
```

Reply:
I'm afraid we can't reuse the compose structure. I agree with @Zheaoli ’s option, the `ServiceConfig` is complicated for users. Another reason is that even if the parameters have the same name, they may not have the same type in different scenarios, e.g. `platform` only needs to be a string in the `run` command, but it needs to be a []string in the `build` command because it needs to support parsing parameters like `arm64,amd64`

Reply:
Hi @AkihiroSuda, I'd like to do some double-check to make sure I understand your suggestion correctly:

`[default_run_config]` will be decoded as `ServiceConfig` and `[default_build_config]` will be decoded as `BuildConfig`, right? 

And another question, 
> (Some fields such as image and container_name cannot be supported and should raise an error.)

which means If the user adds these unsupported fields to the `[default_run_config]` section of the `Nerdctl.toml` file, we should raise an error when decoding?

Reply:
> [default_run_config] will be decoded as ServiceConfig and [default_build_config] will be decoded as BuildConfig, right?

ServiceConfig contains BuildConfig, so no need to have "default_build_config".
A single "default_template" (per namespace) will suffice.

> which means If the user adds these unsupported fields to the [default_run_config] section of the Nerdctl.toml file, we should raise an error when decoding?

Yes. At least a warning should be printed.
Like this: https://github.com/containerd/nerdctl/search?q=reflectutil.UnknownNonEmptyFields